cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -172,7 +173,8 @@ protected void validateOptions(Program program, ProgramOptions options) {
    *                once the program is launch.
    */
   protected abstract void setupLaunchConfig(ProgramLaunchConfig launchConfig, Program program, ProgramOptions options,
-                                            CConfiguration cConf, Configuration hConf, File tempDir) throws IOException;
+                                            CConfiguration cConf, Configuration hConf, File tempDir) throws IOException,
+                                            ExecutionException, InterruptedException;
```

Comment:
Is there a need to surface the execution and interrupted exception? How the caller handle it differently than the IOException?